File: 0258-add-digits/0258-add-digits.java
class Solution {
    public int addDigits(int num) {
        if(num < 10) return num;
        int sum = 0;
        int temp = num;

        while(num >= 10){
            int rem = temp % 10;
            sum += rem;
            temp /= 10;

            if(temp == 0){
                num = sum;
                temp = sum;
                sum = 0;
            }
        }
        return num;
    }
}

File: 0258-add-digits/0258-add-digits.kt
class Solution {
    fun addDigits(num: Int): Int {
        var n = num

        while(n > 9){
            var res = 0
            while(n > 0){
                val temp = n % 10
                res += temp
                n /= 10
            }
            n = res
        }

        return n
    }
}